src/components/Trainer/Trainer.js

```diff
@@ -1,7 +1,327 @@
+			[224, 224]
+		);
+	}
+
+	screenshot = () => {
+		return this.webcam.current.webcam.current.getScreenshot();
+	};
+
+	addExample() {
+		this.setState({busy: true});
+		this.model.addExample(
+			this.screenshot,
+			() => this.capture(this.webcam.current.webcam.current.video),
+			this.state.item,
+			this.state.burstCount
+		);
+	}
+
+	addFromFile = () => {
+		this.setState({busy: true});
+
+		this.images = [];
+		this.fileIndex = 0;
+		let index = 0;
+		const reader = new FileReader();
+
+		reader.addEventListener('load', result => {
+			this.images.push(result.target.result);
+			if (index < this.files.current.files.length) {
+				reader.readAsDataURL(this.files.current.files[index++]);
+			} else {
+				this.model.addExample(
+					() => this.images[this.fileIndex++],
+					this.captureFromFile,
+					this.state.item,
+					this.files.current.files.length
+				);
+			}
+		});
+
+		reader.readAsDataURL(this.files.current.files[index++]);
+	};
+
+	train() {
+		this.setState({busy: true});
+		this.model.train(
+			this.state.hiddenUnits,
+			this.state.batchSizeFraction,
+			this.state.learningRate,
+			this.state.epochs,
+			this.state.setSize,
+			this.state.randomness
+		);
+	}
+
+	predict() {
+		if (this.webcam.current.webcam.current) {
+			this.setState({busy: true});
+			this.model.predict(
+				this.capture(this.webcam.current.webcam.current.video)
+			);
+		} else {
+			this.setStatus('Please connect a camera.');
+		}
+	}
+
+	getName(item) {
+		return item.name + (item.qualifier ? ` (${item.qualifier})` : '');
+	}
+
+	setStatus(status) {
+		this.setState({status});
+		if (status.indexOf('...') === -1) this.setState({busy: false});
+	}
+
+	render() {
+		return (
+			<div>
+				<div className="col" style={{textAlign: 'center'}}>
+					<span id="status-text">{this.state.status}</span>
+					<br />
+					<WebcamCapture
+						setPrediction={() => {}}
+						history={{push: () => {}}}
+						ref={this.webcam}
+					/>
+					<br />
+					<ItemSelector
+						item={this.state.item}
+						items={Object.values(this.model.items)}
+						setItem={item => this.setState({item})}
+						disabled={this.state.busy}
+					/>
+					<br />
+					<label>Burst Count:</label>
+					<input
+						type="text"
+						value={this.state.burstCount}
+						size={3}
+						maxLength={3}
+						disabled={this.state.busy}
+						onChange={e =>
+							this.setState({burstCount: e.target.value})
+						}
+					/>
+					<button
+						onClick={() =>
+							this.addExample(
+								this.webcam.current.webcam.current.video
+							)
+						}
+						disabled={
+							this.state.busy ||
+							!this.webcam.current.webcam.current
+						}>
+						Add From Camera
+					</button>
+					<br />
+					<input
+						type="file"
+						multiple
+						ref={this.files}
+						disabled={this.state.busy}
+					/>
+					<button
+						onClick={this.addFromFile}
+						disabled={this.state.busy}>
+						Add From File
+					</button>
+				</div>
+
+				<div className="col">
```

Comment:
Could reuse a component here instead of 6 nearly identical input fields to make it more DRY